pulser-core/pulser/sampler/sampler.py

```diff
@@ -28,8 +28,20 @@ def sample(
         raise NotImplementedError("Parametrized sequences can't be sampled.")
 
     samples_list = []
+    if seq._slm_mask_dmm:
+        seq.insert_slm_mask()
```

Comment:
I'm not sure I understand why this is being added at sampling time. Imo, sampling should not alter the contents of the sequence and it seems that this does that

Reply:
Also, isn't this being added every time the sequence is sampled?